src/python/pants/backend/python/rules/pex.py

```diff
@@ -248,12 +261,22 @@ def log(self, *args, **kwargs) -> None:
     else:
         argv.append("--no-manylinux")
 
+    argv.append("--no-compile")
+
     if request.entry_point is not None:
         argv.extend(["--entry-point", request.entry_point])
 
     if python_setup.requirement_constraints is not None:
         argv.extend(["--constraints", python_setup.requirement_constraints])
 
+    argv.extend(
+        [
+            "--no-pypi",
+            *(f"--index={url}" for url in python_repos.indexes),
+            *(f"--find-links={url}" for url in python_repos.repos),
+        ]
+    )
```

Comment:
I'm going to defer to others (John) on if this part makes sense. It looks right to me, but I'm not very familiar with it.

Reply:
This is correct but that is not knowable locally. It is probably worth a comment that this works iff PythonRepos.indexes defaults to PyPI - which it currently does; otherwise, it is broken.